third_party/nvidia/lib/TritonNVIDIAGPUToLLVM/MemoryOpToLLVM.cpp

```diff
@@ -137,63 +138,109 @@ struct LocalLoadOpConversion
 };
 
 LogicalResult lowerDistributedToSharedStmatrix(
-    Location loc, TypedValue<RankedTensorType> src, MemDescType memDescType,
-    Value adaptorSrc, Value smemBase, const TypeConverter *typeConverter,
-    ConversionPatternRewriter &rewriter, const TargetInfoBase &targetInfo,
+    Location loc, RankedTensorType tensorTy, MemDescType memDescType,
+    Value adaptorSrc, Value smemBase, Type llvmElemTy,
+    ConversionPatternRewriter &rewriter, const TargetInfo &targetInfo,
     std::pair<size_t, Type> *const llvmOpCount = nullptr) {
+  if (!targetInfo.supportLdStMatrix())
+    return failure();
+
+  assert(llvmOpCount == nullptr && "NYI");
   auto b = TritonLLVMOpBuilder(loc, rewriter);
-  auto mmaEncoding =
-      dyn_cast<triton::gpu::NvidiaMmaEncodingAttr>(src.getType().getEncoding());
-  if (!mmaEncoding)
+  auto *ctx = tensorTy.getContext();
+  auto regL = toLinearLayout(tensorTy.getShape(), tensorTy.getEncoding());
+  auto memL = toLinearLayout(memDescType.getShape(), memDescType.getEncoding());
+  auto cvt = minimalCvtLayout(memDescType, tensorTy);
+
+  auto S = [ctx](StringRef v) { return StringAttr::get(ctx, v); };
+  auto kReg = S("register");
+  auto kLane = S("lane");
+  auto kWarp = S("warp");
+  auto kBlock = S("block");
+  auto kOffset = S("offset");
+  auto smemPtrTy = ptr_ty(ctx, 3);
+
+  // Just stmatrix for now
+  // 1) NYI in the stmatrix lowering
+  //    Pack everything into uint32_t to support bitwidths other than 16
+  auto bitwidth = tensorTy.getElementTypeBitWidth();
+  if (bitwidth != 16)
     return failure();
-  auto sharedLayout =
-      dyn_cast<triton::gpu::NVMMASharedEncodingAttr>(memDescType.getEncoding());
-  if (!sharedLayout)
+
+  // Inter block stmatrix is not supported
+  if (cvt.hasInDim(kBlock))
```

Comment:
cc @peterbell10
There was a test in `test_tensor_descriptor.py` that was not passing with 2CTAs. Now with this line passes. This line says that we just bail out if there's any interCTA business going on. It seems to work just fine as long as you are addressing things within your own CTA, even without any map or anything.